### PR TITLE
Upgrade to Django 4.2

### PR DESCRIPTION
**Pull request by willbarton**
Django 4.2 support was added in #8188, this PR upgrades the base version, removes the one 3.2 shim we had to put in place, and updates documentation links to all point to `stable` rather than a specific version (except where a specific version is warranted). 

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)

### CODE REVIEW

File: cfgov/cfgov/settings/base.py [hunk 1]
```diff
@@ -20,7 +20,27 @@
...
+try:
+    SECRET_KEY = os.environ["SECRET_KEY"]
+except KeyError:
+    raise ImproperlyConfigured(
+        "The SECRET_KEY environment variable must be set"
+    )
+
+# Secret key fallbacks that allow the rotation of the secret key.
+# This environment variable should be a JSON array if fallbacks are available.
+# This list if empty (no old fallback secret keys) if the environment variable
```
Comment (chosak) [inline cfgov/cfgov/settings/base.py:33]:
```suggestion
# This list is empty (no old fallback secret keys) if the environment variable
```

File: cfgov/cfgov/settings/production.py [hunk 2]
```diff
@@ -185,8 +185,30 @@
...
 SECURE_HSTS_PRELOAD = True
+SECURE_SSL_REDIRECT = True
+
+# Require the SECRET_KEY as an environment variable
+try:
+    SECRET_KEY = os.environ["SECRET_KEY"]
+except KeyError:
+    raise ImproperlyConfigured(
+        "The SECRET_KEY environment variable must be set"
+    )
```
Comment (chosak) [inline cfgov/cfgov/settings/production.py:200]: Currently this passes if `SECRET_KEY` is set but empty, e.g. `SECRET_KEY= cfgov/manage.py ...`.

```suggestion
SECRET_KEY = os.getenv("SECRET_KEY")

if not SECRET_KEY:
    raise ImproperlyConfigured(
        "The SECRET_KEY environment variable must be set"
    )
```

or if you really want to be "modern":

```py
if not (SECRET_KEY := os.getenv("SECRET_KEY")):
   ...
```

File: cfgov/cfgov/settings/production.py [hunk 3]
```diff
@@ -185,8 +185,30 @@
...
+
+# Secret key fallbacks that allow the rotation of the secret key.
+# This environment variable should be a JSON array if fallbacks are available.
+# This list is empty (no old fallback secret keys) if the environment variable
+# SECRET_KEY_FALLBACKS is not set.
+try:
```
Comment (chosak) [inline cfgov/cfgov/settings/production.py:206]: Would it be nice to unify this logic with the `ALLOWED_HOSTS` logic above, by creating a simple helper function? Would avoid some duplication.

Comment (willbarton) [reply cfgov/cfgov/settings/production.py:206]: This is a very good idea, and would apply to `CORS_ALLOWED_ORIGINS` too!

Comment (willbarton) [reply cfgov/cfgov/settings/production.py]: [Added](https://github.com/cfpb/consumerfinance.gov/pull/8196/files#diff-399c82f449c78fc61059580ebcc2436b6d010ee9b6e0d3e87603a53e1d86c770R18-R26) in eb373065854a345921e00d240d13cac788a1882f !

**Review by chosak — APPROVED** [review]
image

**Comment by anselmbradford** [comment]
kraken

**Pull request merged**